R/cluster.R

```diff
@@ -0,0 +1,92 @@
+#' Clustering wrapper
+#'
+#' This function returns a \code{SummarizedExperiment} with clustering 
+#'   information in its colData or rowData
+#'
+#' @param x A
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'
+#' @param BLUSPARAM A 
+#'   \code{\link[https://rdrr.io/github/LTLA/bluster/man/BlusterParam-class.html]{BlusterParam}}
+#'   object.
+#'
+#' @param assay.type A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to use for identifying dominant taxa.
+#'
+#' @param assay_name a single \code{character} value for specifying which
+#'   assay to use for calculation.
+#'   (Please use \code{assay.type} instead. At some point \code{assay_name}
+#'   will be disabled.)
+#'  
+#'  @param MARGIN A single numeric value for selecting if the clustering is done
+#'   row-wise / for features (1) or column-wise / for samples (2). Must be \code{1} or
+#'   \code{2}. (By default: \code{MARGIN = 1}) 
+#'   
+#' @param full A logical scalar indicating whether the clustering statistics 
+#'   from both steps should be put in metadata
+#'   
+#' @details
+#' When setting \code{full=TRUE}, the clustering information will be stored in
+#' the metadata of the object.
+#'
+#' @author Basil Courbayre
+#'
+#' @examples
+#' data(GlobalPatterns)
+#' x <- GlobalPatterns
+#'
+#' # Cluster on rows using Kmeans
+#' tse <- cluster(tse, KmeansParam(centers = 3))
+#'
+NULL
+
+#' @rdname cluster
+#' @export
+setGeneric("cluster", signature = c("x"),
+           function(x, BLUSPARAM, assay.type = assay_name, assay_name = "counts", 
+                    MARGIN=1, full = FALSE)
+               standardGeneric("cluster"))
+
+#' @rdname cluster
+#' @export
+#' @importFrom bluster clusterRows
+setMethod("cluster", signature = c(x = "SummarizedExperiment"),
+    function(x, BLUSPARAM, assay.type = assay_name, assay_name = "counts", 
+             MARGIN=1, full = FALSE) {
+        # Checking parameters
```

Comment:
name parameter would be also nice so that user can specify what is the name of the data that will be stored to metadata. Check for instance DMM from mia (I guess it has this implemented).

Also: what if the name is already used? I think the function should not overwrite

Reply:
Should it display an error? Or append a number to the name and proceed? I think I could also do a prompt to insert a new name but that may not be the best idea since I haven't seen any prompt in R so far.

Personally, I prefer the error as the numbers can be confusing and it's easy to get lost in them.

Reply:
Error might be safer than using a different name than user was providing.

Reply:
Should I also provide a name for the colData/rowData?

Reply:
Yes, add name parameter. --> if error, then the name should be checked in the beginning of the function (it would not be nice surprise to wait long time just to see this error)